modules/ROOT/pages/lb-mapping-rules.adoc

```diff
@@ -8,165 +8,278 @@ image:logo-hybrid-disabled.png[link="/runtime-manager/deployment-strategies", ti
 |===
 
-This rule maps `lb-demo.lb.anypointdns.net/{app}` to `{app}.cloudhub.io`. `{app}` is a pattern for any application name you pass in the inbound request.
 
+== Using Patterns in Mapping Rules
 
-=== Host Mapping
+A _pattern_ is a string that defines a template for matching an input text. The value in curly brackets (`{   }`) is treated as a variable.
 
-If you have a wildcard certificate (such as `*.example.com`), you can use the `subdomain` variable to map any subdomain:
+* Variable *names* can contain only lowercase letters (`a-z`) and no other characters.
+* Variable *values* can contain the characters `a-z0-1.&?-_`, but can't contain slash characters (`/`).
 
-[%header,cols="10a,20a,20a,10a"]
+
+[IMPORTANT]
+DLBs don't support patterns in the *Output Path* (`appURI`).
+
+If you set up a DNS CNAME record to map `example.com` to `_lb-name_.lb.anypointdns.net`,
+you can then use a mapping rule to map `app.example.com` to a different deployed CloudHub Mule application name.
+
+[%header,cols="10a,20a,20a,20a,10a"]
 |===
-|Rule # |Input URL 2+^| Output URL
-|   |  *URI*   |       *appName*    |  *appURI*
-| 0 | /        | {subdomain}   | /
+| Rule # | *Input Path* | *Target App* | *Output Path* | *Protocol*
+|1 | /api/v1/ | default-app | /api/v1/ | http
+|2 | /api/v1/secondapp | second-app | /api/v1	http
 |===
-This rule automatically maps any request passed to a subdomain of `example.com` to the corresponding `appName`:
 
-* Passing `api.example.com` would redirect to `api.cloudhub.io` +
-* Passing `application.example.com` is mapped to `application.cloudhub.io`.
 
-The same applies to the https://en.wikipedia.org/wiki/Subject_Alternative_Name[Subject Alternative Names] (SANs) of your SSL endpoints.
+You can route external requests to `+https://app.example.com+` to `+http://app.cloudhub.io/example+`:
 
-Instead of using a wildcard certificate, which permits any subdomain application name to be passed through the dedicated load balancer, you can restrict the allowed subdomain names. To do this, you must have different SANs configured for a certificate's common name. You can use the `subdomain` variable to map the subdomain portion of your domain name to your application.
+[%header,cols="10a,20a,20a,20a,10a"]
+|===
+| Rule # | *Input Path* | *Target App* | *Output Path* | *Protocol*
+|1 | app.example.com | app | /example | http
+|===
 
-In this example:
+*DID I SCREW THIS UP COMPLETELY?*
 
-* Two deployed applications:
-** `dev-app`
-** `qa-app`
-* An SSL endpoint with the SANs:
-** `dev.example.com`
-** `qa.example.com`
-* Mapping rule:
-+
-[%header,cols="10a,20a,20a,10a"]
+Alternatively, you can define a pattern to hold the input value:
+
+[%header,cols="10a,20a,20a,20a,10a"]
 |===
-|Rule # |Input URL 2+^| Output URL
-|   |  *URI*   |       *appName*    |  *appURI*
-| 0 | /        | {subdomain}-app   | /
+| Rule # | *Input Path* | *Target App* | *Output Path* | *Protocol*
+|1 | example.com/{myPattern} | app-{myPattern} | /data | http
```

Comment:
|1 | /{myPattern}/ | app-{myPattern} | /data | http

Where did you get this off? It's repeated several times. I don't know if it's the view that's showing extra stuff or not.

Reply:
This is from the existing docs:
> Or you can define a pattern to hold the input value:
> ‘example.com/{mypattern}’ -> application: `app-{mypattern}` URI: /data
> The example above causes that both example.com/bookings and example.com/sales match to app-bookings/data and app-sales/data respectively, as the variable mypattern holds these values. For input=”bookings.example.com”, the pattern can be resolved by assigning mypattern=”bookings” and for input=`sales.example.com`, the pattern is resolved to assign mypattern=”sales”.

Reply:
Not sure what you mean by repeated several times - I'm not seeing that in the source.

Reply:
Looks like it's gone now... previously when loading the diff view it was multiplicating entire paragraphs.